src/sql/functions/totp_functions.sql

```diff
@@ -0,0 +1,155 @@
+-- NAMESPACE: totp
+-- REQUIRES: user
+
+---
+---  TOTP Functions
+---
+
+-- Create a new TOTP setup for a user
+CREATE OR REPLACE FUNCTION create_totp_setup(_user_id integer, _secret text)
+ RETURNS void AS $$
+
+    INSERT INTO user_totp (user_id, secret, verified)
+    VALUES (_user_id, _secret, FALSE)
+    ON CONFLICT (user_id) DO UPDATE
+    SET secret = EXCLUDED.secret,
+        verified = FALSE,
+        created_at = NOW(),
```

Comment:
```suggestion
```

I don't think we need this line for the `ON CONFLICT` clause since the column will already have been created. We should use the `updated_at` column to check and see if a time based secret has expired.

Reply:
Updated!